app/services/versioning/base_updater.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Versioning
+  class BaseUpdater
+    attr_reader :revision, :user
+
+    def initialize(revision, user)
+      @revision = revision
+      @user = user
+    end
+
+    def column_names
+      raise "Implement with e.g. 'revision.class.column_names'"
```

Comment:
Super minor as no-one will see these exceptions but I don't think this quite reads right. As if you substitute e.g. with for example it's like "Implement with for example .." which doesn't feel like correct English. 

I don't mind if you include an example of what they need to implement but I don't think that's needed. As if we can suggest it maybe we should have just implemented it as a default? Just an error saying it's not implemented is likely enough.

Reply:
Fair enough - although I think the English was fine :-).